x/blockdb/memory_db.go

```diff
@@ -0,0 +1,79 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package blockdb
+
+import "sync"
+
+// MemoryDatabase is an in-memory implementation of BlockDB
+type MemoryDatabase struct {
```

Comment:
If we moved this to its own package this could be called `Database` since there'd be no collision against the existing `Database` type